src/Model/State.re

```diff
@@ -80,13 +79,34 @@ type t = {
 };
 
 let initial = (~getUserSettings, ~contributedCommands, ~workingDirectory) => {
-  let editorGroups = EditorGroups.create();
-  let initialEditorGroup = editorGroups |> EditorGroups.getFirstEditorGroup;
+  Vim.init();
+
+  let initialBuffer = {
+    let Vim.BufferMetadata.{id, version, filePath, modified, _} =
+      Vim.Buffer.openFile(BufferPath.welcome) |> Vim.BufferMetadata.ofBuffer;
+    Buffer.ofMetadata(~id, ~version, ~filePath, ~modified);
+  };
```

Comment:
It seems misleading that creating an initial state should have these sort of side-effects. Can you think of a better place for this to live?

Reply:
Agreed, pulling this out into an `init` function somewhere and passing it in seems like a good idea.

Reply:
That sounds like a good solution - thanks @glennsl !

Reply:
Fixed in https://github.com/onivim/oni2/pull/1977/commits/b19acca26fef78f0c80cc633d1aa15b3be419cde

Reply:
Thanks @glennsl !